drivers/analog/adc.c

```diff
@@ -756,18 +758,47 @@ int adc_register(FAR const char *path, FAR struct adc_dev_s *dev)
   DEBUGASSERT(dev->ad_ops->ao_reset != NULL);
   dev->ad_ops->ao_reset(dev);
 
+  /* Malloc for af_channale and af_data */
+
+  fifosize = fifo->af_fifosize;
+  if (fifosize == 0)
```

Comment:
If `fifo->af_fifosize` is nonzero (I suppose this is set from BSP/arch during ADC initialization, right?), the BSP/arch is also responsible for allocating `af_channel` and `af_data`? What if this doesn't happen? I would let `adc_register` do all the allocation and just set default size if `fifo->af_fifosize = 0`.

Reply:
how about skip allocation of af_channel/af_data if it isn't zero?

Reply:
When fifosize is not 0, af_data and channel are passed from the lower half of the vendor.The reason why fifosize is equal to 0 is that the vendor layer does not define the fifosize macros of each ADC device separately. Therefore, nuttx still uses CONFIG_ADC_FIFOSIZE uniformly. If the fifosize of each ADC is defined separately on the vendor side, then the judgment of if (fifosize == 0) will not be entered.

Reply:
My point was more about what happens if the lower half does not initialize `af_data` and `af_channel` correctly. There should at least be some check at the end of `adc_register` that controls whether both arrays are allocated imho.

Reply:
thanks，I understand what you mean, and will make some changes

Reply:
where the change?

Reply:
done